### PR TITLE
:seedling: move all kustomize manifests from `go/v3` to `kustomize/v1`

### PR DESCRIPTION
**Pull request by estroz**
The `kustomize/v1` plugin should contain all kustomize-related manifests, and `go/v3` should contain only go-related files.

Signed-off-by: Eric Stroczynski <ericstroczynski@gmail.com>

### CODE REVIEW

File: pkg/plugins/common/kustomize/v1/scaffolds/api.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		); err != nil {
+			return fmt.Errorf("error scaffolding APIs: %v", err)
+		}
+
+		if err := scaffold.Execute(
+			&crd.Kustomization{},
+			&crd.KustomizeConfig{},
+		); err != nil {
+			return fmt.Errorf("error scaffolding kustomization: %v", err)
+		}
```
Comment (Adirio) [inline pkg/plugins/common/kustomize/v1/scaffolds/api.go:91]: Does it still make sense to keep them two different `Execute` calls with 2 error messages?

File: pkg/plugins/common/kustomize/v1/create.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	resource *resource.Resource
+
+	flagSet *pflag.FlagSet
+
+	// force indicates whether to scaffold files even if they exist.
+	force bool
+}
+
+func (p *createSubcommand) UpdateMetadata(plugin.CLIMetadata, *plugin.SubcommandMetadata) {}
+func (p *createSubcommand) BindFlags(fs *pflag.FlagSet)                                   { p.flagSet = fs }
```
Comment (Adirio) [inline pkg/plugins/common/kustomize/v1/create.go:40]: Reason?

File: pkg/plugins/common/kustomize/v1/create.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	config   config.Config
+	resource *resource.Resource
+
+	flagSet *pflag.FlagSet
+
+	// force indicates whether to scaffold files even if they exist.
+	force bool
+}
+
+func (p *createSubcommand) UpdateMetadata(plugin.CLIMetadata, *plugin.SubcommandMetadata) {}
```
Comment (Adirio) [inline pkg/plugins/common/kustomize/v1/create.go:39]:
```suggestion
```

Not defining it is valid. They are optional methods.

File: pkg/plugins/common/kustomize/v1/scaffolds/api.go (hunk 1 above)
Comment (estroz) [reply pkg/plugins/common/kustomize/v1/scaffolds/api.go:91]: Probably not.

File: pkg/plugins/common/kustomize/v1/create.go (hunk 2 above)
Comment (estroz) [reply pkg/plugins/common/kustomize/v1/create.go:40]: So `Scaffold()` can grab flag values after they have been set in the CLI.

**Comment by camilamacedo86** [comment]
The changes show great for me 👍 
/lgtm

/hold
to give a chance for we ensure that https://github.com/operator-framework/operator-sdk/pull/4768 first.

File: pkg/plugins/common/kustomize/v1/create.go (hunk 2 above)
Comment (Adirio) [reply pkg/plugins/common/kustomize/v1/create.go]: Yeah but looked for usages and found none.

**Pull request merged**